build-tools/packages/build-tools/src/fluidBuild/buildGraph.ts

```diff
@@ -60,12 +61,13 @@ class TaskStats {
 	public leafQueueWaitTimeTotal = 0;
 }
 
-class BuildContext {
+class BuildContextInternal {
 	public readonly fileHashCache = new FileHashCache();
 	public readonly taskStats = new TaskStats();
 	public readonly failedTaskLines: string[] = [];
 	constructor(
 		public readonly repoPackageMap: Map<string, Package>,
+		public readonly context: BuildContext,
```

Comment:
This is confusing. Can't rename either one more fully? If renaming `BuildContextInternal` more, then also rename where it is stored as `buildContext`.

Reply:
Agreed. I renamed it further to "statsContext" and now pass the buildContext separately, so there are two contexts. I don't love that option but it is clearer that before. I noticed when I renamed the statsContext that 

I think some of it could be combined/reorganized or extend a common context interface for the buildGraph-specific context. I'll explore that next, but wanted to push the changes as they are in case the other idea doesn't go anywhere.

Reply:
In the latest iteration, I renamed the statsContext to BuildGraphContext and it implements BuildContext. I renamed the properties on BuildGraph and BuildPackage to the generic `context` since it now contains everything from both the more widely used BuildContext and the BuildGraphContext.